be/src/storage/lake/starlet_location_provider.cpp

```diff
@@ -6,40 +6,38 @@
 
 #include <fmt/format.h>
 
+#include "common/logging.h"
 #include "fs/fs_starlet.h"
+#include "gutil/strings/util.h"
 #include "service/staros_worker.h"
 
 namespace starrocks::lake {
 
-static std::string normalize_group(const std::string& group) {
-    if (group.back() != '/') {
-        return group;
-    }
-    return group.substr(0, group.length() - 1);
+std::string StarletLocationProvider::root_location(int64_t tablet_id) const {
+    return fmt::format("{}?ShardId={}", kStarletPrefix, tablet_id);
 }
 
-StatusOr<std::string> StarletLocationProvider::root_location(int64_t tablet_id) {
-    if (g_worker == nullptr) {
-        return Status::InternalError("init_staros_worker() must be called before get_shard_info()");
-    }
-    ASSIGN_OR_RETURN(auto shard_info, g_worker->get_shard_info(tablet_id));
-    return normalize_group(shard_info.obj_store_info.s3_obj_store.uri);
+std::string StarletLocationProvider::tablet_metadata_location(int64_t tablet_id, int64_t version) const {
+    return fmt::format("{}/tbl_{:016X}_{:016X}?ShardId={}", kStarletPrefix, tablet_id, version, tablet_id);
```

Comment:
kStarletPrefix => "staros://", this will assemble a path like `staros:///tbl_`, triple slash (/) there

Reply:
Fixed, PTAL